### PR TITLE
[Feature] llama index 0.10.x migration

### PR DESCRIPTION
**Pull request by gaya3-zipstack**
## What

Migrating to llama-index 0.10.x

## Why

To stay updated with the latest llama-index library

## How

Making required code changes and deprecation of service context

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

- https://docs.llamaindex.ai/en/latest/getting_started/v0_10_0_migration.html
- https://docs.llamaindex.ai/en/latest/module_guides/supporting_modules/service_context_migration.html

## Related Issues or PRs

- Unstract adapters PR - https://github.com/Zipstack/unstract-adapters/pull/19
- Unstract SDK PR - https://github.com/Zipstack/unstract-sdk/pull/19

## Dependencies Versions

-

## Notes on Testing

- Tested classifier tool
- Tested prompt studio indexing and answering prompt

## Screenshots
Test Connection LLM
image

Test Connection Vector DB

image

Test Connection  Embedding

image

Test COnnection X2Text

image

Prompt Studio Profile
image

image

Indexing
image

Prompt answer

image

Classifier - tool testing

Settings
image

File upload
image

Tool run with output
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/pyproject.toml [hunk 1]
```diff
@@ -30,8 +30,8 @@ dependencies = [
     "python-socketio==5.9.0", # For log_events
     "social-auth-app-django==5.3.0", # For OAuth
     "social-auth-core==4.4.2", # For OAuth
-    "unstract-sdk~=0.18.0",
-    "unstract-adapters~=0.8.0",
+    "unstract-sdk==0.19.0",
```
Comment (chandrasekharan-zipstack) [inline backend/pyproject.toml:33]: NIT: Any reason for pinning this and not leaving it as `~=`

File: prompt-service/src/unstract/prompt_service/main.py [hunk 2]
```diff
@@ -300,8 +297,13 @@ def prompt_processor() -> Any:
                 "Failed due to vector db error",
             )
             return result, 500
+        # Set up llm, embedding and callback manager to collect usage stats
+        # for this context
+        UNCallbackManager.set_callback_manager(
```
Comment (chandrasekharan-zipstack) [inline prompt-service/src/unstract/prompt_service/main.py:302]: NIT: We might need other params to track the usage to the prompt studio project. Can be taken up through a separate PR

**Review by chandrasekharan-zipstack — APPROVED** [review]
LGTM - make sure its merged only after the packages are published and the lock files are updated

File: unstract/core/src/unstract/core/llm_workflow_generator/llm_interface.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import uuid
 
 import redis
-from llama_index.llms import AzureOpenAI
+from llama_index.llms.azure_openai import AzureOpenAI
```
Comment (hari-kuriakose) [inline unstract/core/src/unstract/core/llm_workflow_generator/llm_interface.py:9]: @gaya3-zipstack Should this be `from llama_index.core.llms.azure_openai import AzureOpenAI`?

File: .gitignore [hunk 4]
```diff
@@ -641,8 +641,8 @@ backend/backend/settings/*
 !backend/backend/settings/test.py
 
 # Local Dependencies for docker testing
- unstract/unstract-adapters/
- unstract/unstract-sdk/
+unstract/unstract-adapters/
+unstract/unstract-sdk/
```
Comment (hari-kuriakose) [inline .gitignore:645]: @gaya3-zipstack Let's review the Docker setup once.

File: backend/pyproject.toml (hunk 1 above)
Comment (gaya3-zipstack) [reply backend/pyproject.toml:33]: Thank you for catching. Was trying out a few things and forgot to change it here. Done now.

File: tools/classifier/src/main.py [hunk 5]
```diff
@@ -8,11 +8,15 @@
...
     LogState,
     MetadataKey,
+    ToolEnv,
     ToolSettingsKey,
 )
 from unstract.sdk.llm import ToolLLM
 from unstract.sdk.tool.base import BaseTool
 from unstract.sdk.tool.entrypoint import ToolEntrypoint
+from unstract.sdk.utils.callback_manager import (
+    CallbackManager as UNCallbackManager,
```
[the remark this answers is not in the document]
Comment (gaya3-zipstack) [reply tools/classifier/src/main.py:18]: llamaIndex has a CallbackManager too @hari-kuriakose . Just to differentiate and avoid confusion to the reader of the code, this practice is being followed.

**Review by nehabagdia — APPROVED** [review]
